storage/src/time_stamp_comparator.cpp

```diff
@@ -10,6 +10,11 @@ namespace rocksdb {
 
 using namespace ::rocksdb;
 
+Slice ExtractTimestampFromUserKey(const Slice& user_key, size_t ts_sz) {
+  ConcordAssert(user_key.size() >= ts_sz);
+  return Slice(user_key.data() + user_key.size() - ts_sz, ts_sz);
+}
+
 class Lexicographic64TsComparator : public Comparator {
  public:
   Lexicographic64TsComparator() : Comparator(/*ts_sz=*/sizeof(uint64_t)), cmp_without_ts_(BytewiseComparator()) {
```

Comment:
As this is implementing Comparator interface, we should add a virtual dtor:
virtual ~ Lexicographic64TsComparator() {}